server/app/views/v1/stacks/_stack.json.jbuilder

```diff
@@ -15,5 +15,6 @@ json.variables latest_rev.variables
 json.services stack.grid_services.to_a do |grid_service|
   json.partial! 'app/views/v1/grid_services/grid_service', grid_service: grid_service
 end
-json.parent stack.has_parent? ? { name: stack.parent_name } : nil
-json.children stack.children.pluck(:name).map { |child_name| { name: child_name } }
+
+json.parent stack.parent.try(:to_relation)
+json.children stack.children.map(&:to_relation)
```

Comment:
I think we want to have the views match the resulting JSON API schema as explicitly as possible, and thus avoid defining parts of the JSON schema in the models. It's more important to be able to see what the JSON API schema is from reading the views, even at the cost of a couple lines of code duplication for the `name` and `id`. IOW this should be something like in https://github.com/kontena/kontena/pull/2866#discussion_r141277250

AFAIK there's only a couple places in the views where the structure of the JSON depends on what's in the models, like the grid `{ "stats": ... }`, and even those are already sometimes a pain.